src/libraries/System.Private.CoreLib/src/System/Reflection/Nullability/NullabilityInfoContext.cs

```diff
@@ -0,0 +1,482 @@
+            var getterAttributes = propertyInfo.GetGetMethod(true)?.ReturnParameter.GetCustomAttributesData();
+            var setterAttributes = propertyInfo.GetSetMethod(true)?.GetParameters()[0].GetCustomAttributesData();
+
+            if (getterAttributes != null)
+            {
+                CheckNullabilityAttributes(nullability, getterAttributes);
+            }
+
+            if (setterAttributes != null)
+            {
+                CheckNullabilityAttributes(nullability, setterAttributes);
+            }
+
+            return nullability;
+        }
+
+        /// <summary>
+        /// Populates <see cref="NullabilityInfo" /> for the given <see cref="EventInfo" />.
+        /// </summary>
+        /// <param name="eventInfo">The parameter which nullability info gets populated</param>
+        /// <exception cref="ArgumentNullException">If the eventInfo parameter is null</exception>
+        /// <returns><see cref="NullabilityInfo" /></returns>
+        [RequiresUnreferencedCode("Nullability attributes are trimmed by the linker")]
+        public NullabilityInfo Create(EventInfo eventInfo)
+        {
+            if (eventInfo is null)
+            {
+                throw new ArgumentNullException(nameof(eventInfo));
+            }
+
+            return GetNullabilityInfo(eventInfo, eventInfo.EventHandlerType!, eventInfo.GetCustomAttributesData());
+        }
+
+        /// <summary>
+        /// Populates <see cref="NullabilityInfo" /> for the given <see cref="FieldInfo" />.
+        /// </summary>
+        /// <param name="fieldInfo">The parameter which nullability info gets populated</param>
+        /// <exception cref="ArgumentNullException">If the fieldInfo parameter is null</exception>
+        /// <returns><see cref="NullabilityInfo" /></returns>
+        [RequiresUnreferencedCode("Nullability attributes are trimmed by the linker")]
+        public NullabilityInfo Create(FieldInfo fieldInfo)
+        {
+            if (fieldInfo is null)
+            {
+                throw new ArgumentNullException(nameof(fieldInfo));
+            }
+
+            if (fieldInfo.IsPrivate || fieldInfo.IsFamilyAndAssembly || fieldInfo.IsAssembly)
+            {
+                if (IsPublicOnly(fieldInfo.IsPrivate, fieldInfo.IsFamilyAndAssembly, fieldInfo.IsAssembly, fieldInfo.Module))
+                {
+                    return new NullabilityInfo(fieldInfo.FieldType, NullabilityState.Unknown, NullabilityState.Unknown, null, s_emptyArray);
+                }
+            }
+
+            var attributes = fieldInfo.GetCustomAttributesData();
+            var nullability = GetNullabilityInfo(fieldInfo, fieldInfo.FieldType, attributes);
+            CheckNullabilityAttributes(nullability, attributes);
+            return nullability;
+        }
+
+        private bool IsPublicOnly(bool isPrivate, bool isFamilyAndAssembly, bool isAssembly, Module module)
```

Comment:
Should `isFamilyOrAssembly` (aka `protected internal`) also be considered? Are there tests for that?

Reply:
I will test that, as far as i know, protected members was not affected

Reply:
Added some test, could not test exactly as could not find an assembly that removed nullability attribute for the internal member's too, it only applied for private members, for now i just guess that because of `protected` modifier `protected internal` members will not be affected